typhon/widgets.py

```diff
@@ -238,3 +239,63 @@ def channel(self, value):
     def _tx(self, value):
         """Receive information from happi channel"""
         self.add_device(value['obj'])
+
+
+class SignalDialogButton(QPushButton):
+    """QPushButton to launch a QDialog with a PyDMWidget"""
+    text = NotImplemented
```

Comment:
Should these be customizable at `__init__` time as well?

Reply:
Yeah I can add these. It did bother me that the base class was unusable.